### DISCUSSION TITLE
How to solve the net::ERR_CONNECTION_REFUSED Error

### DISCUSSION THREAD

**Discussion by ruanqizhen** [valhalla/valhalla#4324] [Q&A]
I've build a tile server, and I was able to get results with curl.  But when I started the Valhalla-app to test the service, I got this error in the web browser:

```
Access to XMLHttpRequest at 'http://(mypublicip):8002/height' from origin 'http://(mypublicip)' has been blocked by CORS policy:
Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

POST http://(mypublicip):8002/height net::ERR_FAILED
```
|   |   |   |
| ---- | --- | ---- 
| (anonymous) | @ | xhr.js:256 
| xhr | @ | xhr.js:49 
| ai | @ | dispatchRequest.js:51 
| value | @ | Axios.js:146
| (anonymous) | @ | Axios.js:185
| (anonymous) | @ | bind.js:5
| value | @ | Map.jsx:766
| (anonymous) | @ | Map.jsx:225
| fire | @ | Events.js:195
| _fireDOMEvent | @ | Map.js:1452
| _handleDOMEvent | @ | Map.js:1401
| o | @ | DomEvent.js:108

It looks like CORS problem, but I'm not very sure.

Any idea how to fix this problem?

**Comment by nilsnolde** [comment]
Where's the Valhalla server running and how did you run it? What does the browser console say (should clearly scream COOOORS)?

**Reply by nilsnolde** [reply]
Sounds more like 8002 isn't open on localhost. Did you use docker and not open the port to the host?

**Reply by ruanqizhen** [reply]
I initiated the Valhalla server using the "valhalla_service" command on the local machine, specifically on port 8002. Interestingly, I discovered that disabling the web browser's security check resolved the issues, indicating that the Valhalla service itself is functioning correctly. It seems to be a matter of some security settings that caused the problem.
